dashboard/app/models/course.rb

```diff
@@ -355,6 +355,16 @@ def has_older_version_progress?(user)
       count > 0
   end
 
+  # returns whether a script in this course has version_warning_dismissed.
+  def has_dismissed_version_warning?(user)
+    script_ids = default_scripts.map(&:id)
+    !user.
+      user_scripts.
+      where(script_id: script_ids).
+      select(&:version_warning_dismissed).
+      empty?
```

Comment:
nit: instead of `!query.empty?` use `query.any?`

Reply:
Or [`exists?`](http://guides.rubyonrails.org/active_record_querying.html#existence-of-objects) I think? Should generate same/similar SQL, but might be more readable:
```rb
user.user_scripts.exists?(
  script_id: script_ids,
  version_warning_dismissed: true
)
```

Reply:
Unfortunately that won't work because the `version_warning_dismissed` is a serialized property in the `properties` JSON column :(

Once we can support inner-JSON queries that would be a better solution.

Note, line 364 runs in Rails, not in SQL.

Reply:
Oh, bummer!  I didn't realize that.  Since it was called `select` I figured it was still ActiveRecord code, I understand what's going on now.